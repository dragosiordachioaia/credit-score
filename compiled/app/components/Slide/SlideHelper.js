"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDebtSlide = exports.getBalanceSlide = exports.getOffersSlide = exports.getScoreSlide = void 0;

var _react = _interopRequireDefault(require("react"));

var _bem = _interopRequireDefault(require("../../utils/bem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cn = (0, _bem.default)({
  block: 'slide'
});

var getScoreSlide = function getScoreSlide(props, state) {
  var currentSlideData = props.slides[state.currentSlideIndex];
  var crtScoreToDisplay = state.currentCoefficientNumber * currentSlideData.maxScore;
  return _react.default.createElement("div", {
    key: "slide-score",
    className: cn('individual-slide'),
    style: {
      width: "".concat(props.radius * 2, "px")
    }
  }, _react.default.createElement("p", {
    className: cn('small-text')
  }, "Your credit score is"), _react.default.createElement("h2", {
    className: cn('score-text'),
    style: {
      color: props.slides[state.currentSlideIndex].color
    }
  }, Math.round(crtScoreToDisplay)), _react.default.createElement("p", {
    className: cn('small-text')
  }, "out of", _react.default.createElement("b", {
    className: cn('max-score')
  }, state.maxScore)), _react.default.createElement("p", {
    className: cn('description-text'),
    style: {
      color: props.slides[state.currentSlideIndex].color
    }
  }, "Soaring high"));
};

exports.getScoreSlide = getScoreSlide;

var getOffersSlide = function getOffersSlide(props, state) {
  var currentSlideData = props.slides[state.currentSlideIndex];
  var crtScoreToDisplay = state.currentCoefficientNumber * currentSlideData.maxScore;
  return _react.default.createElement("div", {
    key: "slide-offers",
    className: cn('individual-slide'),
    style: {
      width: "".concat(props.radius * 2, "px")
    }
  }, _react.default.createElement("h2", {
    className: cn('score-text'),
    style: {
      color: props.slides[state.currentSlideIndex].color
    }
  }, Math.round(crtScoreToDisplay)), _react.default.createElement("p", {
    className: cn('small-text')
  }, "New offers"));
};

exports.getOffersSlide = getOffersSlide;

var getBalanceSlide = function getBalanceSlide(props, state) {
  return _react.default.createElement("div", {
    key: "slide-offers",
    className: cn('individual-slide'),
    style: {
      width: "".concat(props.radius * 2, "px")
    }
  }, _react.default.createElement("img", {
    src: "/cards.png",
    className: cn('cards')
  }), _react.default.createElement("p", {
    className: cn('small-text')
  }, "Transfer your ", _react.default.createElement("br", null), " balance!"));
};

exports.getBalanceSlide = getBalanceSlide;

var getDebtSlide = function getDebtSlide(props, state) {
  var currentSlideData = props.slides[state.currentSlideIndex];
  var crtScoreToDisplay = state.currentCoefficientNumber * currentSlideData.maxScore;
  var changeText = "\xA3".concat(Math.abs(currentSlideData.change));

  if (currentSlideData.change < 0) {
    changeText = "-".concat(changeText);
  }

  return _react.default.createElement("div", {
    key: "slide-debt",
    className: cn('individual-slide'),
    style: {
      width: "".concat(props.radius * 2, "px")
    }
  }, _react.default.createElement("p", {
    className: cn('small-text')
  }, "Your long term debt is"), _react.default.createElement("h2", {
    className: cn('score-text'),
    style: {
      color: props.slides[state.currentSlideIndex].color
    }
  }, "\xA3", Math.round(crtScoreToDisplay)), _react.default.createElement("p", {
    className: cn('small-text')
  }, "Change since last check: ", _react.default.createElement("b", null, changeText)), _react.default.createElement("p", {
    className: cn('description-text'),
    style: {
      color: props.slides[state.currentSlideIndex].color
    }
  }, "You're doing great!"));
};

exports.getDebtSlide = getDebtSlide;