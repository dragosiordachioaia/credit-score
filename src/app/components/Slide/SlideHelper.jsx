import React from 'react';
import bemHelper from '../../utils/bem';

const cn = bemHelper({ block: 'slide' });

export const getScoreSlide = (props, state) => {
  const currentSlideData = props.slides[state.currentSlideIndex];
  const crtScoreToDisplay = state.currentCoefficientNumber * currentSlideData.maxScore;

  return (
    <div
      key='slide-score'
      className={cn('individual-slide')}
      style={{width: `${props.radius * 2}px`}}
    >
      <p className={cn('small-text')}>
        Your credit score is
      </p>
      <h2
        className={cn('score-text')}
        style={{color: props.slides[state.currentSlideIndex].color}}
      >
        {Math.round(crtScoreToDisplay)}
      </h2>
      <p className={cn('small-text')}>
        out of
        <b className={cn('max-score')}>{state.maxScore}</b>
      </p>
      <p
        className={cn('description-text')}
        style={{color: props.slides[state.currentSlideIndex].color}}
      >
        Soaring high
      </p>
    </div>
  )
}

export const getOffersSlide = (props, state) => {
  const currentSlideData = props.slides[state.currentSlideIndex];
  const crtScoreToDisplay = state.currentCoefficientNumber * currentSlideData.maxScore;

  return (
    <div
      key='slide-offers'
      className={cn('individual-slide')}
      style={{width: `${props.radius * 2}px`}}
    >
      <h2
        className={cn('score-text')}
        style={{color: props.slides[state.currentSlideIndex].color}}
      >
        {Math.round(crtScoreToDisplay)}
      </h2>
      <p className={cn('small-text')}>
        New offers
      </p>
    </div>
  )
}

export const getBalanceSlide = (props, state) => {
  return (
    <div
      key='slide-offers'
      className={cn('individual-slide')}
      style={{width: `${props.radius * 2}px`}}
    >
      <img src='/cards.png' className={cn('cards')}></img>
      <p className={cn('small-text')}>
        Transfer your <br /> balance!
      </p>
    </div>
  )
}

export const getDebtSlide = (props, state) => {
  const currentSlideData = props.slides[state.currentSlideIndex];
  const crtScoreToDisplay = state.currentCoefficientNumber *currentSlideData.maxScore;

  let changeText = `£${Math.abs(currentSlideData.change)}`;
  if(currentSlideData.change < 0) {
    changeText = `-${changeText}`;
  }

  return (
    <div
      key='slide-debt'
      className={cn('individual-slide')}
      style={{width: `${props.radius * 2}px`}}
    >
      <p className={cn('small-text')}>
        Your long term debt is
      </p>
      <h2
        className={cn('score-text')}
        style={{color: props.slides[state.currentSlideIndex].color}}>
        £{Math.round(crtScoreToDisplay)}
      </h2>
      <p className={cn('small-text')}>
        Change since last check: <b>{changeText}</b>
      </p>
      <p
        className={cn('description-text')}
        style={{color: props.slides[state.currentSlideIndex].color}}
      >
        You're doing great!
      </p>
    </div>
  )
}
