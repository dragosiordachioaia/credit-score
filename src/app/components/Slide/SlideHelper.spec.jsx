import React from 'react';
import renderer from 'react-test-renderer';
import TweenMax from 'gsap/TweenMax';

import * as helperMethods from './SlideHelper';

const state = {
  currentCoefficientNumber: 0.7342857142857143,
  currentCoefficientStroke: 0.7325868639732143,
  currentSlideIndex: 0,
  maxScore: 700,
  targetScore: 514,
};
const props = {
  "animate": true,
  "size": "big",
  "radius": 150,
  "slides":[
    {"type":"score","score":514,"maxScore":700,"color":"#a7ced1"},
    {"type":"debt","score":24682,"maxScore":49364,"color":"#FCD29F","change":-327},
    {"type":"balance","score":24682,"maxScore":49364,"color":"#FCD29F"},
    {"type":"offers","score":24682,"maxScore":49364,"color":"#FCD29F"},
    {"type":"debt","score":24682,"maxScore":49364,"color":"#FCD29F","change":327},
    {"type":"debt","score":24682,"maxScore":49364,"color":"#FCD29F","change":0},
    {"type":"debt","score":75,"maxScore":100,"color":"#FCD29F","change":0},
  ],
};

describe('Slide helpers', () => {
  it('getScoreSlide renders correctly', () => {
    let currentState = Object.assign({}, state);
    const result = helperMethods.getScoreSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
  it('getOffersSlide renders correctly', () => {
    let currentState = Object.assign({}, state, {currentSlideIndex: 3});
    const result = helperMethods.getOffersSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
  it('getBalanceSlide renders correctly', () => {
    let currentState = Object.assign({}, state, {currentSlideIndex: 2});
    const result = helperMethods.getBalanceSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
  it('getDebtSlide renders correctly when change is negative', () => {
    let currentState = Object.assign({}, state, {currentSlideIndex: 1});
    const result = helperMethods.getDebtSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
  it('getDebtSlide renders correctly when change is positive', () => {
    let currentState = Object.assign({}, state, {currentSlideIndex: 4});
    const result = helperMethods.getDebtSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
  it('getDebtSlide renders correctly when no change', () => {
    let currentState = Object.assign({}, state, {currentSlideIndex: 5});
    const result = helperMethods.getDebtSlide(props, currentState);
    expect(result).toMatchSnapshot();
  });
});
