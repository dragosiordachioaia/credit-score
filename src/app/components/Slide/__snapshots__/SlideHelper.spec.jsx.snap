// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide helpers getBalanceSlide renders correctly 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <img
    className="slide__cards"
    src="/cards.png"
  />
  <p
    className="slide__small-text"
  >
    Transfer your 
    <br />
     balance!
  </p>
</div>
`;

exports[`Slide helpers getDebtSlide renders correctly when change is negative 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <p
    className="slide__small-text"
  >
    Your long term debt is
  </p>
  <h2
    className="slide__score-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    £
    36247
  </h2>
  <p
    className="slide__small-text"
  >
    Total credit limit 
  </p>
  <p
    className="slide__description-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    Down £327 from last month
  </p>
</div>
`;

exports[`Slide helpers getDebtSlide renders correctly when change is positive 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <p
    className="slide__small-text"
  >
    Your long term debt is
  </p>
  <h2
    className="slide__score-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    £
    36247
  </h2>
  <p
    className="slide__small-text"
  >
    Total credit limit 
  </p>
  <p
    className="slide__description-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    Up £327 from last month
  </p>
</div>
`;

exports[`Slide helpers getDebtSlide renders correctly when no change 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <p
    className="slide__small-text"
  >
    Your long term debt is
  </p>
  <h2
    className="slide__score-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    £
    36247
  </h2>
  <p
    className="slide__small-text"
  >
    Total credit limit 
  </p>
  <p
    className="slide__description-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    No change since last month
  </p>
</div>
`;

exports[`Slide helpers getOffersSlide renders correctly 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <h2
    className="slide__score-text"
    style={
      Object {
        "color": "#FCD29F",
      }
    }
  >
    36247
  </h2>
  <p
    className="slide__small-text"
  >
    New offers
  </p>
</div>
`;

exports[`Slide helpers getScoreSlide renders correctly 1`] = `
<div
  className="slide__individual-slide"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <p
    className="slide__small-text"
  >
    Your credit score is
  </p>
  <h2
    className="slide__score-text"
    style={
      Object {
        "color": "#a7ced1",
      }
    }
  >
    514
  </h2>
  <p
    className="slide__small-text"
  >
    out of
    <b
      className="slide__max-score"
    >
      700
    </b>
  </p>
  <p
    className="slide__description-text"
    style={
      Object {
        "color": "#a7ced1",
      }
    }
  >
    Soaring high
  </p>
</div>
`;
